zio-http-datastar-sdk/src/main/scala/zio/http/datastar/ServerSentEventGenerator.scala

```diff
@@ -0,0 +1,195 @@
+  case object Prepend extends ElementPatchMode
+
+  /** Insert at end inside target */
+  case object Append extends ElementPatchMode
+
+  /** Insert before target element */
+  case object Before extends ElementPatchMode
+
+  /** Insert after target element */
+  case object After extends ElementPatchMode
+
+  /** Remove target element from DOM */
+  case object Remove extends ElementPatchMode
+}
+
+final case class PatchElementOptions(
+  selector: Option[CssSelector] = None,
+  mode: ElementPatchMode = ElementPatchMode.Outer,
+  useViewTransition: Boolean = false,
+  eventId: Option[String] = None,
+  retryDuration: Duration = 1000.millis,
+)
+
+object PatchElementOptions {
+  val default: PatchElementOptions = PatchElementOptions()
+}
+
+final case class PatchSignalOptions(
+  onlyIfMissing: Boolean = false,
+  eventId: Option[String] = None,
+  retryDuration: Duration = 1000.millis,
+)
+
+object PatchSignalOptions {
+  val default: PatchSignalOptions = PatchSignalOptions()
+}
+
+final case class ExecuteScriptOptions(
+  autoRemove: Boolean = true,
+  attributes: Seq[(String, String)] = Seq.empty,
+  eventId: Option[String] = None,
+  retryDuration: Duration = 1000.millis,
+)
+
+object ExecuteScriptOptions {
+  val default: ExecuteScriptOptions = ExecuteScriptOptions()
+}
+
+object ServerSentEventGenerator {
+
+  def executeScript(
+    script0: String,
+    options: ExecuteScriptOptions = ExecuteScriptOptions(),
+  ): ZIO[Datastar, Nothing, Unit] = {
+    val removeAttr = if (options.autoRemove) Dom.attr("data-effect", "el.remove") else Dom.empty
+    patchElements(
+      script(Dom.text(script0), removeAttr)(options.attributes.map(a => Dom.attr(a._1, a._2)): _*),
+      PatchElementOptions(eventId = options.eventId, retryDuration = options.retryDuration),
+    )
+  }
+
+  def patchElements(
+    elements: String,
+    options: PatchElementOptions,
+  ): ZIO[Datastar, Nothing, Unit] =
+    patchElements(elements.split('\n').map(Dom.raw).toList, options)
+
+  def patchElements(
+    elements: String,
+  ): ZIO[Datastar, Nothing, Unit] =
+    patchElements(elements, PatchElementOptions.default)
+
+  def patchElements(
+    element: Dom,
+    options: PatchElementOptions,
+  ): ZIO[Datastar, Nothing, Unit] = patchElements(Iterable(element), options)
+
+  def patchElements(
+    element: Dom,
+  ): ZIO[Datastar, Nothing, Unit] =
+    patchElements(Iterable(element), PatchElementOptions.default)
+
+  def patchElements(
+    elements: Iterable[Dom],
+  ): ZIO[Datastar, Nothing, Unit] =
+    patchElements(elements, PatchElementOptions.default)
+
+  def patchElements(
+    elements: Iterable[Dom],
+    options: PatchElementOptions,
+  ): ZIO[Datastar, Nothing, Unit] = {
+
+    val sb = new StringBuilder
+
+    options.selector.foreach(s => sb.append("selector ").append(s.render).append("\n"))
+
+    if (options.mode != ElementPatchMode.Outer) {
+      sb.append("mode ").append(options.mode.toString).append("\n")
```

Comment:
See https://github.com/zio/zio-http/pull/3649/files#r2297058022

```suggestion
      sb.append("mode ").append(options.mode.render).append("\n")
```